### ISSUE TITLE
Landmark taxa in the dynamic hierarchy

### ISSUE THREAD

**Issue by KatjaSchulz** [EOL/eol_website#5]
Opendata is currently experiencing difficulties, so I will share the Mammal branch with the landmark taxa with Jeremy through my Dropbox.  As soon as opendata is back up, I will add a link here.

The revised mammal branch file contains a new column (Landmark) which has values (1,2 or 3) for landmark taxa.

I imagine that we will use these landmark designations as follows:

1. Minimal contexts, e.g., search results:

Display the landmark with value 1 that is the closest ancestor of the target taxon.  We should probably use the common names of landmark taxa when available.

Examples:
Carnivores > Procyon lotor
Bats > Myotis lucifugus
Mammals > Tapirus pinchaque

2. Abbreviated context, e.g., taxon cover

For taxa that are descendents of families, display the family as well as landmarks with values 1 and 2.  For taxa that are not descendents of families, display landmarks with values 1 and 2. 
 
Examples:
Animals > Vertebrates > Mammals > Carnivores > Procyonidae > Procyon lotor
Animals > Vertebrates > Mammals > Bats > Vespertilionidae > Myotis lucifugus
Animals > Vertebrates > Mammals > Tapiridae > Tapirus pinchaque

3. Extended context.  Not sure we actually need this, but I have included the value 3 landmarks just in case we come up with a scenario where we want to have a more detailed taxonomic context without representing every single ancestor of a node.  This view includes some of the major phylogenetic landmarks that are not familiar to the general public but would be appreciated by advanced students and biodiversity nerds.

For taxa that are descendents of families, display the family as well as landmarks with values 1, 3 and 3.  For taxa that are not descendents of families, display landmarks with values 1, 2 and 3. 

Examples:
Life > Eukaryotes > Animals > Deuterostomes > Vertebrates > Tetrapods > Synapsids > Mammals > Carnivores > Procyonidae > Procyon lotor
Life > Eukaryotes > Animals > Deuterostomes > Vertebrates > Tetrapods > Synapsids > Mammals  > Bats > Vespertilionidae > Myotis lucifugus
Life > Eukaryotes > Animals > Deuterostomes > Vertebrates > Tetrapods > Synapsids > Mammals  > Tapiridae > Tapirus pinchaque

4. Full context, e.g., names tab, dynamic hierarchy navigator

Display every step in the hierarchy.

**Comment by KatjaSchulz** [comment]
Looks like opendata is up and running again: [Dynamic Hierarchy Mammal Branch with Landmarks](https://opendata.eol.org/dataset/tram-580-581/resource/889f22d4-798a-4cc9-979a-e76f67177e09)

**Comment by KatjaSchulz** [comment]
Use the canonical name of the taxon in all landmark contexts.

Here are the preliminary instructions for the [autogenerated brief summaries](https://docs.google.com/document/d/1F5hGZv93_BUUde6gEciJ27ePd8_5BFg8ffqDQcrO4t8/edit?usp=sharing):

1. Taxa of rank species

[name] ([common name]) is a[n] [A1] in the family [A2]. 

A1: Use the landmark with value 1 that is the closest ancestor of the species.  Use the English vernacular name, if available, else use the canonical.

A2: Use the name of the family (i.e., not a landmark taxon) of the species.  Use the English vernacular name, if available, else use the canonical.  -- Complication: some family vernaculars have the word "family" in then, e.g., Rosaceae is the rose family.  In that case, the vernacular would make for a very awkward sentence. It would be great if we could implement a rule, use the English vernacular, if available, unless it has the string "family" in it.

We may have a few species that don't have a family in their ancestry.  In those cases, shorten the taxonomy sentence:

[name] ([common name]) is a[n] [A1].

2. Taxa of rank genus

[name] ([common name]) is a genus in the [A1] family [A2]. 

A1: Use the landmark with value 1 that is the closest ancestor of the genus.  Use the English vernacular name, if available, else use the canonical.

A2: Use the name of the family (i.e., not a landmark taxon) of the genus.  Use the English vernacular name, if available, else use the canonical.  -- Complication: some family vernaculars have the word "family" in then, e.g., Rosaceae is the rose family.  In that case, the vernacular would make for a very awkward sentence. It would be great if we could implement a rule, use the English vernacular, if available, unless it has the string "family" in it.

We may have a few genera that don't have a family in their ancestry.  In those cases, shorten the taxonomy sentence:

[name] ([common name]) is a genus in the [A1] 

3. Taxa of rank family

[name] ([common name]) is a family of [A1].

A1: Use the landmark with value 1 that is the closest ancestor of the family.  Use the English vernacular name, if available, else use the canonical.

This will look a little funny for those families with "family" vernaculars, but I think it's still acceptable, e.g., 

Rosaceae (rose family) is a family of plants.

**Comment by KatjaSchulz** [comment]
Update of the taxonomy components of the autogenerated brief summaries:

1. Taxa of rank species

There was a problem with the previous instructions for these sentences.  The proposed structure did not work with most A1 vernaculars because they are plurals (and we want them to be plurals).  I have revised the sentence structure (also fixed in the instructions above) to work with both scientific names (if no vernaculars are available) and plural vernaculars: 

[name] ([common name]) is a species of [A1] in the family [A2].

This is also a bit simpler since it doesn't require the "a(n)" article switch.

Currently, the "in the family [A2]" part of the sentence does not seem to grab taxa of appropriate rank, i.e., families.  I suspect that this is due to the dynamic hierarchy missing rank issue. If there is no family associated with a species (this may happen sometimes, especially in microbes), omit that part of the sentence and end with a full stop after A1.

2. Taxa of rank genus

We should change the genus sentence structure to match that of the species (also fixed in the instructions above):

[name] ([common name]) is a genus of [A1] in the family [A2].

Also, it looks like the scientific name of the genus is added in parentheses after the canonical (examples: https://beta.eol.org/pages/3833346, https://beta.eol.org/pages/46559142).  This should be the vernacular instead.  If there is no vernacular, don't add the parentheses.

There's the same problems with the "in the family [A2]" part not using taxa of rank family.

3. Taxa of rank family

[name] ([common name]) is a family of [A1].

I couldn't find any family pages where A1 was expressed as a vernacular, although many of the A1 should have vernaculars.  Example: https://beta.eol.org/pages/5500. Actinopterygii has had a vernacular name since the pbdb harvest which was a while ago.

Finally, it would be good if the vernaculars from the [English Vernaculars for Landmark Taxa resource](https://beta.eol.org/resources/476) could always be set to PREFERRED automagically. The landmark vernacular data set can be expected to gel after a while. So if it is too much trouble to integrate the sourcing of PREFERRED vernaculars into the code, we could probably get away with running a script for this every once in a while.

Action items for Jeremy:

1. Change the sentence structure for species and genus
2. If this is not already done: add code for an alternative ending of species and genus sentences when there is no family in the species and genus ancestry.
3. Fix dynamic hierarchy missing rank issues
4. Make sure vernaculars are used whenever available.
5. Set all the vernacular from the [English Vernaculars for Landmark Taxa resource](https://beta.eol.org/resources/476) to PREFERRED.

**Comment by jhammock** [comment]
This sounds reasonable to me. Jeremy, if you can ballpark how much effort
the remainder looks like, that will make some of us feel better.

:)

Jen

**Comment by JRice** [comment]
Those are pretty minimal changes, if I captured them correctly:
- There's no "a or an" for a species
- Genus changes to "is a genus in the [A1]of [A1] "
- is a genus in the [A1]of [A1].
...Then that's just a few minutes' time.

The rank issues are an ongoing problem and with the server down I'm blocked, but I probably need a day (not of WORK but of TIME to run scripts) to resolve.

The vernaculars are being used whenever possible; the code is already there. The problem is that it requires "preferred" vernaculars. 

Setting resource 476's vernaculars to preferred is also quick. Maybe 30 min. ...I'm a *little* worried that I had already gone over all of the pages and set *something* to preferred for each (just randomly picking the first one for each language), but that work appears to have been ... lost?  So I'd like a little additional time to check the publishing code to ensure it's not removing the preferred names or something. ...And perhaps to re-run that code (I pasted it somewhere, so should be simple enough to resurrect).  So, maybe a few hours in all.

So, all told, maybe 3-4 hours of *work*, plus a day or so of *time*.  ...Gated by access to the servers, because of the current technical problem, sigh. :|
